### PR TITLE
Candent recipes RoR web app - Full app PR

### PR DESCRIPTION
**Pull request by mullenlucas**
### Important note:
To set up the project, we used a `.env` file which is included in the gitignore, so it is not downloaded in the repository. Said file contains the variables assignations for the psql database.
The user has to create a `.env` file in the root folder with the following format:

```
POSTGRES_HOST='localhost'
POSTGRES_USER='username'  # <-- replace here
POSTGRES_PASSWORD='password'  # <-- replace here
POSTGRES_DEV_DB='recipe_ror_app_dev'
POSTGRES_TEST_DB='recipe_ror_app_test'
POSTGRES_PROD_DB='recipe_ror_app_prod'
# PROD_PASSWORD = 'production_password'
```
(The implementation is specified in the `README` file)

## Implemented features:

- Created a kanban board to assign tasks and advance the project in an organized way (See Projects section)
- Used different branches to implement each of the project's features

* Implemented:

  - A login page
  - A registration page
  - A food list
  - A recipe list
  - A shopping list view for recipes
  - A list of all public recipes from other users

All implementations were done according to the [project guidelines](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md)

## Some resulting pictures

rspecfeat

mainfeat

recipefeat

foodfeat

### CODE REVIEW

File: app/views/recipes/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  </div>
+  
+<div class ="edith-button">
+  <button class="back1"><%= link_to "Edit this recipe", edit_recipe_path(@recipe) %></button>
+  <br>
+  <button class="back1"><%= link_to "Back to recipes", recipes_path %></button>
+<br>
+  <%= button_to "Destroy this recipe", @recipe, method: :delete %>
+</div>
```
Comment (elisha2kyakpo1) [inline app/views/recipes/show.html.erb]: - [x] The requirement for the recipe show page is that users should be allowed to add ingredients.
- [x] On the same recipe show page (when a user clicks a recipe item) should lead to the form that allows the user to add new food, that's if they are the owners of the recipe.
- [x] The recipe show page should allow users to generate a shopping list. Could you fix all these requirements?

recipe

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Tests passing👏


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipes/show.html.erb (hunk 1 above)
Comment (mullenlucas) [reply app/views/recipes/show.html.erb]: Thank you for the accurate observations ! The issues have been addressed

File: app/views/recipes/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
```
Comment (rloterh) [inline app/views/recipes/show.html.erb]: - [x] The Food table shown under the recipe detail page is missing. Kindly ensure that the food table is displayed under each recipe, as in can be seen in the [wireframe design](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/recipe-app/images/recipe_food.png?token=GHSAT0AAAAAABW7GI45NILK6L4GVBGZ3IZSY5Z5W7A):

**Your design:**        |  **The Original design:**
:-------------------------:|:-------------------------:
Your desgin |  Home Page

File: README.md [hunk 3]
```diff
@@ -1,2 +1,202 @@
-# recipe-ror-app
-Using ROR to create a cooking app that tracks inventories, ingredients, and recipes. Users may create recipes, manage their saved information, build shopping lists based on what they have and what they need to complete a dish, and save ingredients.
+<a name="readme-top"></a>
+
+<div align="center">
+
+  <img src="murple_logo.png" alt="logo" width="140"  height="auto" />
```
Comment (rloterh) [inline README.md]: - [x] OPTIONAL
Kindly ensure to remove the Microverse template logo. Optionally, you can replace it with your own logo if you have one.

File: app/views/recipes/show.html.erb (hunk 1 above)
Comment (rloterh) [reply app/views/recipes/show.html.erb]: needs attention!

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Balkhi & Lucas, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 🚦 🚦   🏃 🏃‍♂️ 

## Highlights 🎯 
- Nice job implementing required changes!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly check the inline comments marked **[OPTIONAL]** under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @mullenlucas , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**